### PR TITLE
Show a progress bar for plugin installation during pulumi up

### LINKED ISSUE #3033: Pulumi up silently installs a plugin with no visible progress
I cloned one of the examples and ran `pulumi up` on it. I also happened to be on a slow internet connection.

```
$ pulumi up
Please choose a stack, or create a new one: <create a new stack>
Please enter your desired stack name.
To create a stack in an organization, use the format <org-name>/<stack-name> (e.g. `acmecorp/dev`): dev
Created stack 'dev'
Previewing update (dev):
Moving plugin... done.
```

`Previewing update (dev):` was showing for quite a long time without any visible progress. Based on `Moving plugin... done.` label (that we added recently) I can tell that this time was (partially?) spent on downloading a provider plugin.

Shall we show the download progress here the same way we show it during `install`?

### PR DESCRIPTION
**Pull request by mikhailshilkov**
Fixes #3033

### CODE REVIEW

File: pkg/workspace/plugins.go [hunk 1]
```diff
@@ -598,6 +600,26 @@ func SelectCompatiblePlugin(
...
 
+// ReadCloserProgressBar displays a progress bar for the given closer and returns a wrapper closer to manipulate it.
+func ReadCloserProgressBar(closer io.ReadCloser, size int64, colorization colors.Colorization) io.ReadCloser {
+	if size == -1 {
+		return closer
+	}
+
+	// If we know the length of the download, show a progress bar.
+	bar := pb.New(int(size))
+	bar.Prefix(colorization.Colorize(colors.SpecUnimportant + "Downloading plugin:"))
```
Comment (pgavlin) [inline pkg/workspace/plugins.go]: Feels like we should parameterize the message here

File: pkg/engine/plugins.go [hunk 2]
```diff
@@ -163,10 +165,14 @@ func installPlugin(plugin workspace.PluginInfo) error {
 
 	logging.V(preparePluginVerboseLog).Infof(
 		"installPlugin(%s, %s): initiating download", plugin.Name, plugin.Version)
-	stream, _, err := plugin.Download()
+	stream, size, err := plugin.Download()
 	if err != nil {
 		return err
 	}
+
+	fmt.Printf("[%s plugin %s-%s] installing\n", plugin.Kind, plugin.Name, plugin.Version)
```
Comment (pgavlin) [inline pkg/engine/plugins.go]: Is this `Printf` intentional?

File: pkg/workspace/plugins.go (hunk 1 above)
Comment (mikhailshilkov) [reply pkg/workspace/plugins.go]: In what way?

File: pkg/engine/plugins.go (hunk 2 above)
Comment (mikhailshilkov) [reply pkg/engine/plugins.go]: Yes, to show the name of the plugin being installed (same as during `pulumi plugin install`)

Comment (pgavlin) [reply pkg/engine/plugins.go]: What does this end up looking like in the CLI? Can you provide a demo (e.g. via `asciinema`)?

File: pkg/workspace/plugins.go [hunk 3]
```diff
@@ -598,6 +600,26 @@ func SelectCompatiblePlugin(
 	return bestMatch, nil
 }
 
+// ReadCloserProgressBar displays a progress bar for the given closer and returns a wrapper closer to manipulate it.
+func ReadCloserProgressBar(closer io.ReadCloser, size int64, colorization colors.Colorization) io.ReadCloser {
```
Comment (pgavlin) [inline pkg/workspace/plugins.go]:
```suggestion
func ReadCloserProgressBar(closer io.ReadCloser, size int64, message string, colorization colors.Colorization) io.ReadCloser {
```

File: pkg/workspace/plugins.go (hunk 1 above)
Comment (pgavlin) [reply pkg/workspace/plugins.go]: https://github.com/pulumi/pulumi/pull/3124/files#r316329605

Comment (mikhailshilkov) [reply pkg/workspace/plugins.go]: Got you. Agreed! I started with a less generic function name, and didn't think after renaming. Will do.

File: pkg/engine/plugins.go (hunk 2 above)
Comment (mikhailshilkov) [reply pkg/engine/plugins.go]: That tool doesn't seem to support Windows, but here is the output:

```
$pulumi preview
Previewing update (dev):
[resource plugin azure-1.0.0-beta.1] installing
Downloading plugin: 27.95 MiB / 39.31 MiB [=================>------]  71.10% 18s
```
and when finished
```
$pulumi preview
Previewing update (dev):
[resource plugin azure-1.0.0-beta.1] installing
Downloading plugin: 39.31 MiB / 39.31 MiB [========================] 100.00% 57s
Moving plugin... done.

     Type                                           Name     Plan                                                                            +   pulumi:pulumi:Stack                            aks-dev  create
```

Comment (pgavlin) [reply pkg/engine/plugins.go]: Thanks!

**Pull request merged**